src/main/java/cnlabs/CustomerWebsite/Config/BatchConfiguration.java

```diff
@@ -0,0 +1,137 @@
+import org.springframework.batch.core.annotation.BeforeStep;
+import org.springframework.batch.core.configuration.annotation.EnableBatchProcessing;
+import org.springframework.batch.core.configuration.annotation.JobBuilderFactory;
+import org.springframework.batch.core.configuration.annotation.StepBuilderFactory;
+import org.springframework.batch.core.launch.support.RunIdIncrementer;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.batch.item.ItemReader;
+import org.springframework.batch.item.ItemWriter;
+import org.springframework.batch.item.data.RepositoryItemReader;
+import org.springframework.batch.item.data.builder.RepositoryItemReaderBuilder;
+import org.springframework.batch.item.file.FlatFileItemWriter;
+import org.springframework.batch.item.file.transform.BeanWrapperFieldExtractor;
+import org.springframework.batch.item.file.transform.DelimitedLineAggregator;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+
+@Configuration
+@EnableBatchProcessing
+public class BatchConfiguration {
+
+    // Job
+    @Bean
+    public Job job(JobBuilderFactory jobBuilderFactory, Step step1) {
+        return jobBuilderFactory.get("customer-loader-job")
+                .incrementer(new RunIdIncrementer())
+                .start(step1)
+                .build();
+    }
+
+
+    // Step
+    @Bean
+    public Step step1(StepBuilderFactory stepBuilderFactory, ItemReader<Customer> repositoryReader, DateProcessor processor, CustomerWriter writer) {
+        return stepBuilderFactory.get("step-1")
+                .<Customer, Customer>chunk(100)
+                .reader(repositoryReader)      // EXTRACT
+                .processor(processor)   // TRANSFORM
+                .writer(writer)         // LOAD
+                .build();
+    }
+
+    // ItemReader
+    @Bean
+    public RepositoryItemReader<Customer> repositoryReader(CustomerRepository customerRepository) {
+        return new RepositoryItemReaderBuilder<Customer>()
+                .repository(customerRepository)
+                .methodName("findAll")
+                .sorts(Map.of("id", Sort.Direction.ASC))
+                .name("repository-reader")
+                .build();
+    }
+
+    // ItemProcessor
+    @Component
+    public static class DateProcessor implements ItemProcessor<Customer, Customer> {
+        @Override
+        public Customer process(Customer customer) {
+            if (customer.getBookCheckoutDate() != null) {
+                if (customer.getBookCheckoutDate().plusDays(14).isBefore(LocalDate.now())) {  // if checkout date + 14 < today
```

Comment:
Could be rewritten as one line
```suggestion
            if (customer.getBookCheckoutDate() != null) && customer.getBookCheckoutDate().plusDays(14).isBefore(LocalDate.now())) {  // if checkout date + 14 < today
```